### PR TITLE
P2 removal

### PR DESCRIPTION
**Pull request by xendk**
Tear out P2.

DDBLM-31 DDBLM-32 DDBLM-33

### CODE REVIEW

File: ding2.install [hunk 1]
```diff
@@ -843,3 +843,73 @@ function ding2_update_7073() {
...
+  module_disable($modules);
+  drupal_uninstall_modules($modules);
+}
+
+/**
+ * Revert modules to remove traces of p2.
+ */
+function ding2_update_7076() {
+  // Need to revert variables for these modules to get rid of the
+  // ding_serendipity_info extra field.
```
Comment (kasperg) [inline ding2.install:892]: :+1: nice commenting.

File: ding2.make [hunk 2]
```diff
@@ -77,6 +77,7 @@ projects[dynamic_background][subdir] = "contrib"
 projects[dynamic_background][version] = "2.0-rc4"
 projects[dynamic_background][patch][] = "https://www.drupal.org/files/issues/create_file_path-2410241-1.patch"
 
+; To be removed together with p2.
```
Comment (kasperg) [inline ding2.make:80]: :+1:

File: modules/ding_event/ding_event.features.field_base.inc [hunk 3]
```diff
@@ -364,15 +364,6 @@ function ding_event_field_default_field_bases() {
...
     'field_name' => 'field_ding_event_price',
-    'indexes' => array(),
-    'foreign keys' => array(
-      'format' => array(
-        'columns' => array(
-          'format' => 'format',
-        ),
-        'table' => 'filter_format',
-      ),
-    ),
```
Comment (kasperg) [inline modules/ding_event/ding_event.features.field_base.inc:375]: I wonder why this is removed as  part of this PR?

File: modules/ding_frontpage/ding_frontpage.pages_default.inc [hunk 4]
```diff
@@ -119,62 +119,6 @@ function ding_frontpage_default_page_manager_pages() {
...
-    $pane->position = 2;
-    $pane->locks = array();
-    $pane->uuid = '3f5761b9-1db4-4dc9-87cb-13e33008a1fc';
-    $display->content['new-3f5761b9-1db4-4dc9-87cb-13e33008a1fc'] = $pane;
-    $display->panels['primary'][2] = 'new-3f5761b9-1db4-4dc9-87cb-13e33008a1fc';
-    $pane = new stdClass();
-    $pane->pid = 'new-5e2a0d2d-7b27-498e-a368-d4e9a2888769';
-    $pane->panel = 'primary';
-    $pane->type = 'campaign';
-    $pane->subtype = 'campaign';
```
Comment (kasperg) [inline modules/ding_frontpage/ding_frontpage.pages_default.inc:149]: Do we really want to remove this ding campaign panel pane?

File: modules/ding_staff/ding_staff.features.field_instance.inc [hunk 5]
```diff
@@ -56,7 +56,9 @@ function ding_staff_field_default_field_instances() {
     'widget' => array(
       'active' => 1,
       'module' => 'options',
-      'settings' => array(),
+      'settings' => array(
+        'match_operator' => 'CONTAINS',
+      ),
```
Comment (kasperg) [inline modules/ding_staff/ding_staff.features.field_instance.inc:61]: This change does not seem relevant to the change at hand?

File: modules/ding_staff/ding_staff.views_default.inc [hunk 6]
```diff
@@ -682,13 +678,6 @@ function ding_staff_views_default_views() {
       'override_label' => 1,
       'label' => 'Edit user roles',
     ),
-    'action::views_bulk_operations_user_roles_action' => array(
-      'selected' => 0,
-      'postpone_processing' => 0,
-      'skip_confirmation' => 0,
-      'override_label' => 1,
-      'label' => 'Change user roles',
```
Comment (kasperg) [inline modules/ding_staff/ding_staff.views_default.inc:690]: Is this related to the change at hand?

File: modules/ding_staff/ding_staff.views_default.inc [hunk 7]
```diff
@@ -707,15 +696,6 @@ function ding_staff_views_default_views() {
         ),
       ),
     ),
-    'action::views_bulk_operations_argument_selector_action' => array(
```
Comment (kasperg) [inline modules/ding_staff/ding_staff.views_default.inc:710]: As above.

File: modules/ding_staff/ding_staff.views_default.inc [hunk 8]
```diff
@@ -730,27 +710,6 @@ function ding_staff_views_default_views() {
       'override_label' => 0,
       'label' => '',
     ),
-    'action::flag_user_action' => array(
```
Comment (kasperg) [inline modules/ding_staff/ding_staff.views_default.inc:733]: As above.

File: modules/ding_ting_frontend/ding_ting_frontend.field_group.inc [hunk 9]
```diff
@@ -605,9 +506,9 @@ function ding_ting_frontend_field_group_info() {
     'label' => 'Right column meta',
     'weight' => '1',
     'children' => array(
-      0 => 'ting_title',
-      1 => 'ting_type',
-      2 => 'ting_author',
+      0 => 'ting_author',
+      1 => 'ting_title',
+      2 => 'ting_type',
```
Comment (kasperg) [inline modules/ding_ting_frontend/ding_ting_frontend.field_group.inc:511]: This seems to reorder elements. Can this be right?

File: modules/ding_ting_frontend/ding_ting_frontend.field_group.inc [hunk 10]
```diff
@@ -671,10 +572,9 @@ function ding_ting_frontend_field_group_info() {
     'label' => 'Right column content',
     'weight' => '3',
     'children' => array(
-      0 => 'ting_series',
-      1 => 'ting_abstract',
+      0 => 'ting_abstract',
+      1 => 'ting_series',
```
Comment (kasperg) [inline modules/ding_ting_frontend/ding_ting_frontend.field_group.inc:576]: As above.

File: modules/ding_ting_frontend/ding_ting_frontend.field_group.inc [hunk 11]
```diff
@@ -827,8 +727,8 @@ function ding_ting_frontend_field_group_info() {
     'label' => 'Right column',
     'weight' => '1',
     'children' => array(
-      0 => 'ting_title',
-      1 => 'ting_author',
+      0 => 'ting_author',
+      1 => 'ting_title',
```
Comment (kasperg) [inline modules/ding_ting_frontend/ding_ting_frontend.field_group.inc:731]: As above.

File: modules/ding_ting_frontend/ding_ting_frontend.pages_default.inc [hunk 12]
```diff
@@ -564,36 +564,6 @@ search/ting
     $display->content['new-b00e0b46-6043-48c2-98ad-6446c489e7f9'] = $pane;
     $display->panels['left_sidebar'][0] = 'new-b00e0b46-6043-48c2-98ad-6446c489e7f9';
     $pane = new stdClass();
-    $pane->pid = 'new-26a27e6c-f934-4169-a06e-04db0f61ebf6';
-    $pane->panel = 'left_sidebar';
-    $pane->type = 'campaign';
-    $pane->subtype = 'campaign';
```
Comment (kasperg) [inline modules/ding_ting_frontend/ding_ting_frontend.pages_default.inc:570]: This removed a campaign pane. Is that relevant to the change at hand?

File: modules/p2/ding_interaction/ding_interaction.install [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+ * @file
+ * Install, update and uninstall functions for the ding_interaction module.
+ */
+
+/**
+ * Implements hook_uninstall().
+ */
+function ding_interaction_uninstall() {
+  variable_del('ding_interaction_callbacks');
+  variable_del('ding_interaction_page_count');
```
Comment (kasperg) [inline modules/p2/ding_interaction/ding_interaction.install:13]: 👌

File: modules/p2/ding_message/ding_message.install [hunk 14]
```diff
@@ -60,6 +60,17 @@ function ding_message_schema() {
...
 
+/**
+ * Implements hook_uninstall().
+ */
+function ding_message_uninstall() {
+  variable_get('ding_message_login_delay');
+  variable_get('ding_message_batch_size');
+  variable_get('ding_message_list_check_size');
+  variable_get('ding_message_search_size');
+  variable_get('ding_message_channel_max');
```
Comment (kasperg) [inline modules/p2/ding_message/ding_message.install]: Should these be `variable_del()`?

File: modules/p2/ding_user_consent/ding_user_consent.install [hunk 15]
```diff
@@ -5,6 +5,24 @@
  * Setup the fields needed for ding list.
  */
 
+/**
+ * Implements hook_uninstall().
+ */
+function ding_user_consent_uninstall() {
+  $url = variable_get('ding_user_consent_loan_history_store_description_link', '');
+  if ($url) {
+    list(, $nid) = explode('/', $url);
```
Comment (kasperg) [inline modules/p2/ding_user_consent/ding_user_consent.install:14]: Should be be explicit that we expect/limit `explode` to return 2 elements?

File: modules/p2/ding_user_consent/ding_user_consent.install [hunk 16]
```diff
@@ -5,6 +5,24 @@
...
+    if ($nid) {
+      node_delete($nid);
+    }
+  }
+
+  variable_get('user_consent_activate');
```
Comment (kasperg) [inline modules/p2/ding_user_consent/ding_user_consent.install]:
```suggestion
  variable_del('user_consent_activate');
```

File: modules/p2/ding_user_consent/ding_user_consent.install [hunk 17]
```diff
@@ -5,6 +5,24 @@
...
+  db_delete('variable')
+    ->condition('name', 'ding_user_consent_%', 'like')
+    ->execute();
```
Comment (kasperg) [inline modules/p2/ding_user_consent/ding_user_consent.install:23]: Hmmm - can we do better than this?

File: modules/p2/ting_openlist/ting_openlist.install [hunk 18]
```diff
@@ -58,7 +58,9 @@ function ting_openlist_install() {
   $schema = array();
   ting_openlist_schema_alter($schema);
 
-  db_add_field('users', 'openlist_uid', $schema['users']['fields']['openlist_uid']);
+  // This would be the proper thing to do when uninstalling, but we need the uid
+  // to migrate lists to the new system, so leave it in place.
+  // db_add_field('users', 'openlist_uid', $schema['users']['fields']['openlist_uid']);
```
Comment (kasperg) [inline modules/p2/ting_openlist/ting_openlist.install:63]: :+1:

File: tests/behat/Makefile [hunk 19]
```diff
@@ -11,10 +11,6 @@ setup-modules:
 	$(DRUSH) en -y connie
 	$(DRUSH) dis -y fbs
 
-	# Enable mocking openlist provider.
-	# Currently disabled as the API has changed and Connie haven't been updated.
-	# $(DRUSH) en -y connie_openlist
-
```
Comment (kasperg) [inline tests/behat/Makefile:17]: I wonder if this make file is actually used anymore 😕 .

**Review by kasperg — APPROVED** [review]
:+1: Lots of quality cleaning here.

I have a few questions and I must admit that all the test deletion saddens me a bit but I guess it has be be reimplemented for the new elements.

File: modules/ding_event/ding_event.features.field_base.inc (hunk 3 above)
Comment (xendk) [reply modules/ding_event/ding_event.features.field_base.inc:375]: Changes in how things are dumped in features over time, shows up as cruft when updating features.

Really, whenever any features touching module is updated all features should be updated to check if there's any changes, but that's not happening.

File: modules/ding_frontpage/ding_frontpage.pages_default.inc (hunk 4 above)
Comment (xendk) [reply modules/ding_frontpage/ding_frontpage.pages_default.inc:149]: Yeah, it's old cruft from the old campain module. When campaign_plus took over, nobody bothered to update all the panels, so there's missing panes floating about.

File: modules/ding_staff/ding_staff.features.field_instance.inc (hunk 5 above)
Comment (xendk) [reply modules/ding_staff/ding_staff.features.field_instance.inc:61]: Nope, old features again. I basically have two choices here: Revert the changes I don't think is related to my changes, and hope I don't mess up, and leave the problem to the next guy that needs to implement something. Or accept that there's unrelated changes but at least come out with features that's completely up to date.

I prefer the latter. Scouts rule: leave things better than you found them.

File: modules/ding_ting_frontend/ding_ting_frontend.field_group.inc (hunk 9 above)
Comment (xendk) [reply modules/ding_ting_frontend/ding_ting_frontend.field_group.inc:511]: I think it is... I believe that the order here is irrelevant, it's just a list of which fields belong in the fieldgroup (which sadly seems to be numerically indexed), I think it still uses field weights to determine rendering order.

Note that it has alphabetized the lists.

File: modules/ding_ting_frontend/ding_ting_frontend.pages_default.inc (hunk 12 above)
Comment (xendk) [reply modules/ding_ting_frontend/ding_ting_frontend.pages_default.inc:570]: As I said before, no, but if nobody cleans up...

File: modules/p2/ding_message/ding_message.install (hunk 14 above)
Comment (xendk) [reply modules/p2/ding_message/ding_message.install]: Why, I do believe you're right.

File: modules/p2/ding_user_consent/ding_user_consent.install (hunk 15 above)
Comment (xendk) [reply modules/p2/ding_user_consent/ding_user_consent.install:14]: Well, it's explicit in that we only have two elements in `list()`, it'll just discard the rest. Safeguarding against the url having an extreme number of slashes in a variable set by the module itself seems a bit overkill to me.

File: modules/p2/ding_user_consent/ding_user_consent.install (hunk 17 above)
Comment (xendk) [reply modules/p2/ding_user_consent/ding_user_consent.install:23]: Such as? There's no wildcard support in `variable_del()`.

Comment (kasperg) [reply modules/p2/ding_user_consent/ding_user_consent.install:23]: If these is a large but limited amount of known variables we could delete them one by one.

File: modules/ding_staff/ding_staff.features.field_instance.inc (hunk 5 above)
Comment (kasperg) [reply modules/ding_staff/ding_staff.features.field_instance.inc:61]: OK - but when creating the PR for the final review we have to be explicit about the fact that such changes are included and why we choose to do so.

File: modules/p2/ding_user_consent/ding_user_consent.install (hunk 17 above)
Comment (xendk) [reply modules/p2/ding_user_consent/ding_user_consent.install:23]: The admin form constructs variable names by invoking a provider function and looping over the result, so we don't actually know what names might exist.
